src/sysc/scc/ordered_semaphore.h

```diff
@@ -155,12 +155,13 @@ class SC_API ordered_semaphore : public sc_core::sc_semaphore_if, public sc_core
                     queue[1].pop_front();
                     return false;
                 }
-            } else {
+            } else if(queue[0].size()) {
                 if(queue[0].front() == sc_core::sc_get_current_process_handle()) {
                     queue[0].pop_front();
                     return false;
                 }
-            }
+            } else
```

Comment:
These two lines stem from an earlier (and buggy version). If non of the queues (which set-up 2 priorities) have entries the in_use() function should return true.